Add syscall commands

daniellimws opened the pull request:
## Add syscall commands ##

### Description/Motivation/Screenshots ###
This was mentioned earlier in #257 

<!-- Describe technically what your patch does. -->
<!-- Why is this change required? What problem does it solve? -->
This patch allows gdb to retrieve syscall related information from syscall-tables in gef-extras so that the user can quickly check what is the next syscall instruction going to do without needing to refer externally.
<!-- Why is this patch will make a better world? -->
Hopefully, this can reduce the hassle of needing to open another tab in the browser to refer to syscalls every time we encounter one.
<!-- How does this look? Add a screenshot if you can -->
https://asciinema.org/a/gNwy6khs3rkQAPyv1CMrFAnlf

### How Has This Been Tested? ###
Refer to asciinema above.

| Architecture | Yes/No                   | Comments               |
|--------------|:------------------------:|------------------------|
| x86-32       | :heavy_check_mark:  |                  |
| x86-64       |  :heavy_check_mark:  |                  |
| ARM          | :heavy_multiplication_x: |                        |
| AARCH64      | :heavy_multiplication_x: |                        |
| MIPS         | :heavy_multiplication_x: |                        |
| POWERPC      | :heavy_multiplication_x: |                        |
| SPARC        | :heavy_multiplication_x: |                        |

### Checklist ###

<!-- N.B.: Your patch won't be reviewed unless fulfilling the following base requirements: -->
<!--- Put an `x` in all the boxes that apply. -->
- [x] My code follows the code style of this project.
- [ ] My change includes a change to the documentation, if required.
- [x] I have read and agree to the **CONTRIBUTING** document.
- [ ] I have tagged the PR as appropriate (e.g. bug fix).

daniellimws commented:
I will add documentation tomorrow.

Grazfather commented on gef.py:
```diff
@@ -1571,6 +1573,7 @@ class X86(Architecture):
         21: "identification",
     }
     syscall_register = "$eax"
+    syscall_instructions = ['sysenter', 'int 0x80']
```
nitpick: double quotes

Grazfather commented on gef.py:
```diff
@@ -8151,6 +8160,100 @@ def clean(self, event):
+        p = self.get_settings_path()
+        if not p: return None
+        return os.path.join(p, "{}.py".format(x))
+
```
nit: extra line

Grazfather commented on gef.py:
```diff
@@ -8151,6 +8160,100 @@ def clean(self, event):
+
+        path = self.get_settings_path()
+        if path is None:
+            err("Cannot open '{0}': check directory and/or `gef config {0}` setting, currently: '{1}'".format("syscall-args.path",
```
Long line, let's keep them under 120 chars

Grazfather commented on gef.py:
```diff
@@ -8151,6 +8160,100 @@ def clean(self, event):
+        headers = [Color.colorify(x, attrs=color) for x in ["Parameter", "Register", "Value"]]
+        param_names = [re.split(' |\*', p)[-1] for p in parameters]
+        info("{:<28} {:<28} {}".format(*headers))
+        for i in range(len(parameters)):
```
I think I'd prefer `for name, register, value in zip(param_names, registers, values):` or similar.

daniellimws commented:
Alright fixed those issues

daniellimws commented:
Updated docs as well.
Somehow the descriptions don't show up next to the list of commands when I enter `gef` in gdb. I've added comments at the top of the class of each command, am I doing it right?

Grazfather commented:
The issue with the command listing is that you have a newline at the start of your docstring. This is a bug in gef. Filed #285.
